### PR TITLE
Update to latest go-nitro API with payment support

### LINKED ISSUE #56: Integrate payments into testground script
Depends on statechannels/go-nitro/issues/775

### PR DESCRIPTION
**Pull request by lalexgap**
Fixes #56 

This updates `go-nitro-testground` to use the latest `go-nitro` code (https://github.com/statechannels/go-nitro/commit/e70703cba68dbd58922d95d80442121274bb03c7).

This version of `go-nitro` fully implements payments which means we can defunct the payment client we implemented in this repo.

### CODE REVIEW

File: tests/virtual-payment.go [hunk 1]
```diff
@@ -51,20 +62,27 @@ func CreateVirtualPaymentTest(runEnv *runtime.RunEnv, init *run.InitContext) err
...
 	defer chainSyncer.Close()
 
-	pc := paymentclient.NewClient(me, peers, chainSyncer.ChainService(), runEnv.D())
-	defer pc.Close()
+	store := store.NewMemStore(crypto.FromECDSA(&me.PrivateKey))
 
+	ms := m.NewP2PMessageService(me, peers, runEnv.D())
+
+	// The outputs folder will be copied when results are collected.
+	logDestination, _ := os.OpenFile("./outputs/nitro-client.log", os.O_CREATE|os.O_WRONLY, 0666)
```
Comment (kerzhner) [inline tests/virtual-payment.go:70]: Should we check the error value?

File: tests/virtual-payment.go [hunk 2]
```diff
@@ -81,22 +99,53 @@ func CreateVirtualPaymentTest(runEnv *runtime.RunEnv, init *run.InitContext) err
...
+					Nonce:             rand.Int63(),
+				}
+
+				r := nClient.CreateVirtualChannel(request)
+				channelId = r.ChannelId
+				cm.WaitForObjectivesToComplete([]protocols.ObjectiveId{r.Id})
+
+				runEnv.RecordMessage("Opened virtual channel %s with %s using hub %s", utils.Abbreviate(channelId), utils.Abbreviate(randomPayee.Address), utils.Abbreviate(randomHub.Address))
+
+				nClient.Pay(r.ChannelId, big.NewInt(int64(1)))
```
Comment (kerzhner) [inline tests/virtual-payment.go:134]: Why is a payment made on this line and on line 139?

File: utils/testing.go [hunk 3]
```diff
@@ -114,3 +120,41 @@ func SelectRandom[U ~[]T, T any](collection U) T {
 	randomIndex := rand.Intn(len(collection))
 	return collection[randomIndex]
 }
+
+// CreateLedgerChannels creates a directly funded ledger channel with each hub in hubs.
+// The funding for each channel will be set to amount for both participants.
+// this function blocks until all ledger channels have successfully been created.
```
Comment (kerzhner) [inline utils/testing.go:126]:
```suggestion
// This function blocks until all ledger channels have successfully been created.
```

**Review by kerzhner — APPROVED** [review]

File: tests/virtual-payment.go (hunk 2 above)
Comment (lalexgap) [reply tests/virtual-payment.go:134]: We want to record the duration to the first payment, so the first call to `Pay` is timed. The first call to `Pay` is in the scope of runEnv.D().Timer("time_to_first_payment") that ends on line 136.

Comment (kerzhner) [reply tests/virtual-payment.go:134]: Gotcha, thanks for the explanation!

**Pull request merged**